data_structures/linked_list.cpp

```diff
@@ -1,58 +1,147 @@
+/**
+ * @file linked_list.cpp
+ * @brief Implementation of singly linked list algorithm.
+ *
+ * @details
+ * The linked list is a form of container used for holding a sequence of 
+ * values, which can be added, removed and displayed.
+ * ### Algorithm
+ * Values can be added by iterating to the end of a list(by following 
+ * the pointers) starting from the first node. Whichever node points to null
+ * is considered the last node and is pointed to the new value.
+ *
+ * Values can be removed by also iterating through the list. When the node containing 
+ * the value is found, the node pointing to the current node is made to point to the 
+ * node that the current node is pointing to, and then returning the current node to
+ * heap store.
+ */
 #include <iostream>
+#include <sstream>
 
-struct node {
+/**
+ * @namespace data_structures
+ * @brief Data Structures algorithms
+ */
+namespace data_structures {
+/**
+ * stores value of a node and address of the next node
+ */
+struct node
+{
     int val;
     node *next;
 };
+	
+/**
+ * This function checks if the string passed consists 
+ * of only digits.
+ * @param s To be checked if s contains only integers
+ * @returns true if there are only only digits present in the string
+ * @returns false if any other character is found
+ */
+bool isDigit(const std::string& s)
+{
+	//function statements here
+	for(int i=0;i<s.size()-1;i++)
+    {
+    	if(!isdigit(s[i]))
+    	{
+            return false;
+		}
+	}
+	return true;
+}
+	
+/**
+ * This function changes the value_type of its argument
+ * from a string to an integer
+ * @param s argument whose value_type we want to change
+ * @return number value with value_type as integer
+ */
+int toInt(const std::string& s)
+{
+	//function statements here
+	std::stringstream geek(s);
+	int number=0;
+	geek>>number;
+	return number;
+} 
 
-node *start;
+node *start=nullptr;
```

Comment:
Don't register global variables.
Pass them as local function parameters.

Or even better, register them in a class.